### PR TITLE
Serverless internal docker registry node port - docs

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- add docs to explain how internal docker registry works

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/17459

### CODE REVIEW

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+Internal docker registry is used for serverless to push built function images which can be used to deploy function without using 3rd party service.
+
+The following diagram illustrates how it works.
+
+![Serverless architecture](./assets/svls-internal-registry.svg)
+
+1. Build job pushes function image to docker registry using in-cluster URL
+2. Internal docker registry URL is resolved by K8s DNS to real ip address
+3. Kubelet fetch image using URL: `localhost:{node_port}/{image}`
```
Comment (kwiatekus) [inline docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:13]: I think we should mention also the fact that when multiple nodes are used, the nodePort routing will assure that the request will reach the node where internal registry is deployed

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+title: Internal Docker Registry
+---
+
+Internal docker registry is used for serverless to push built function images which can be used to deploy function without using 3rd party service.
```
Comment (kwiatekus) [inline docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:5]: I'm missing an opening like:
```
Kyma serverless module by default comes with internal docker registry deployment that is used to store function container images w/o necessity to  use 3rd party registry

Internal docker  is not recommended for production, as

is not deployed in HA setup
has limited storage space and no garbage collection of orphaned images
Still, it is very convenient for development and getting first time experience of kyma serverless.
```

**Review by kwiatekus — CHANGES_REQUESTED** [review]

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md (hunk 2 above)
Comment (kwiatekus) [reply docs/05-technical-reference/00-architecture/svls-02-internal-registry.md]: please provide proper punctuation

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+has limited storage space and no garbage collection of orphaned images
+Still, it is very convenient for development and getting first time experience of kyma serverless.
+
+The following diagram illustrates how it works.
+
+![Serverless architecture](./assets/svls-internal-registry.svg)
+
+1. Build job pushes function image to docker registry using in-cluster URL
+2. Internal docker registry URL is resolved by K8s DNS to real ip address
+3. Kubelet fetch image using URL: `localhost:{node_port}/{image}`
```
Comment (grego952) [inline docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:19]:
```suggestion
3. [kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) fetches the image using the URL: `localhost:{node_port}/{image}`.
```

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+4. NodePort allows kubelet to get into the cluster network and translate `localhost` to `internal-registry.kyma-system.svc.cluster.local` and ask the k8s dns to resolve the name
```
Comment (grego952) [inline docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:20]:
```suggestion
4. NodePort allows kubelet to get into the cluster network and translate `localhost` to `internal-registry.kyma-system.svc.cluster.local` and ask the Kubernetes DNS to resolve the name.
```

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+5. K8s DNS service resolve the name and provide the IP of `internal docker registry`
```
Comment (grego952) [inline docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:21]:
```suggestion
5. The Kubernetes DNS service resolves the name and provides the IP of the internal Docker registry.
```
Do we need the `` ticks for the internal Docker registry?

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+**NOTE:** Kubelet cannot resolve in-cluster URL that's why serverless uses NodePort service.
```
Comment (grego952) [inline docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:23]:
```suggestion
**NOTE:** kubelet cannot resolve the in-cluster URL. That's why Serverless uses the NodePort service.
```
Maybe we can move this note higher? Under point 3 or 4?

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+**NOTE:** NodePort service routing assure that pull request reaches the internal docker registry regardless of whether it is from different node.
```
Comment (grego952) [inline docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:24]:
```suggestion
**NOTE:** The NodePort service routing assures that the pull request reaches the internal Docker registry regardless of whether it is from a different node.
```
Maybe we can also move this note higher, where it suits better? Maybe under point 4?

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+---
+title: Internal Docker Registry
+---
+
+Kyma serverless module by default comes with internal docker registry deployment that is used to store function container images w/o necessity to  use 3rd party registry
```
[the remark this answers is not in the document]
Comment (dbadura) [reply docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:5]:
```suggestion
By default, the Kyma Serverless module comes with the internal Docker registry, which stores the Function container images without using the third-party registry.
```

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md (hunk 5 above)
Comment (dbadura) [reply docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:21]: No

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+![Serverless architecture](./assets/svls-internal-registry.svg)
+
+1. Build job pushes the Function image to the Docker registry using the in-cluster URL.
+2. The Kubernetes DNS resolves the internal Docker registry URL to the actual IP address.
+3. [kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) fetches the image using the URL: `localhost:{node_port}/{image}`.
+4. NodePort allows kubelet to get into the cluster network and translate `localhost` to `internal-registry.kyma-system.svc.cluster.local` and ask the Kubernetes DNS to resolve the name.
+5. The Kubernetes DNS service resolves the name and provides the IP of the internal Docker registry.
+
+**NOTE:** kubelet cannot resolve the in-cluster URL. That's why Serverless uses the NodePort service.
```
Comment (dbadura) [inline docs/05-technical-reference/00-architecture/svls-02-internal-registry.md:21]:
```suggestion
**NOTE:** kubelet cannot resolve the in-cluster URL. That's why Serverless uses the NodePort service.

```

File: docs/05-technical-reference/00-architecture/svls-02-internal-registry.md (hunk 7 above)
Comment (dbadura) [reply docs/05-technical-reference/00-architecture/svls-02-internal-registry.md]: I don't have a strong opinion, but I would like to keep it at the bottom to not interfere with the explanation of how it works.

**Review by grego952 — APPROVED** [review]

**Review by kwiatekus — APPROVED** [review]

**Pull request merged**